### PR TITLE
App refactor 1

### PR DESCRIPTION
**Pull request by dherault**
## Summary

🎉 
Introducing the Plural app, 100% TS-typed! Faster, better, stronger!
🎉 
- [x] Convert app to TypeScript
- [x] Move contexts to contexts dir
- [x] Update dependencies
- [x] Edit codegen
- [x] Delete unimported files

👉  To review, run `npx tsc` to check for TS errors (THERE WILL BE NONE 💪 ) and explore the app as you normally would.

## Checklist

- [x] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [x] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: www/src/components/clusters/Clusters.tsx [hunk 1]
```diff
@@ -1,135 +1,354 @@
...
+import moment from 'moment'
+import {
+  Chip,
+  ClusterIcon,
+  ReloadIcon,
+  StatusIpIcon,
+  StatusOkIcon,
+} from 'pluralsh-design-system'
+import { useEffect, useMemo, useState } from 'react'
+
```
Comment (dogmar) [inline www/src/components/clusters/Clusters.tsx:20]: It looks like `Clusters.tsx` got overwritten by a converted `Clusters.js`, which I'm guessing isn't what we want, right?

**Review by dogmar — CHANGES_REQUESTED** [review]
A few small comments here and there, and one major question:

It looks like your bash script blew away `Clusters.tsx` with a converted `Clusters.js`, which I'm guessing is a legacy file that was meant to be removed a while back? Or is this intended?

Also, thanks for taking on this big refactor. I'm excited to have typescript everywhere soon!

File: www/src/components/Invite.tsx [hunk 2]
```diff
@@ -81,15 +81,15 @@ function ExistingInvite({ invite: { account }, id }) {
...
+            onClick={() => mutation()}
             loading={loading}
             width="100%"
             padding="medium"
           >
             Join {account.name}
           </Button>
         </Box>
-      </Box>
+      </Box>mo
```
[the remark this answers is not in the document]
Comment (dherault) [reply www/src/components/Invite.tsx]: Done

File: www/src/components/repository/RepositoryArtifacts.tsx [hunk 3]
```diff
@@ -14,21 +25,30 @@ import InfiniteScroller from '../utils/InfiniteScroller'
...
+      fullColor
+    />
+  ),
+  LINUX: (
+    // @ts-expect-error
+    <UbuntuLogoIcon
+      size="14px"
+      fullColor
+    />
+  ),
```
[the remark this answers is not in the document]
Comment (dherault) [reply www/src/components/repository/RepositoryArtifacts.tsx]: Done

File: www/src/components/shell/CreateRepositoryCard.tsx [hunk 4]
```diff
@@ -36,7 +36,7 @@ function CreateRepositoryCard({ data, onPrevious = () => {} }) {
               // START <<Remove this after dev>>
               if (process.env.NODE_ENV !== 'production' && DEBUG_SCM_TOKENS[provider]) {
                 console.log('going to ', `/oauth/callback/${provider.toLowerCase()}/shell?code=abcd`)
-                window.location = `/oauth/callback/${provider.toLowerCase()}/shell?code=abcd`
+                window.location = `/oauth/callback/${provider.toLowerCase()}/shell?code=abcd` as any as Location
```
[the remark this answers is not in the document]
Comment (dherault) [reply www/src/components/shell/CreateRepositoryCard.tsx]: Done everywhere

File: www/src/components/utils/SmoothScroller.tsx [hunk 5]
```diff
@@ -226,8 +226,20 @@ export default function SmoothScroller({
...
+     loadNextPage,
+      mapper,
+       listRef,
+        setListRef,
+         handleScroll,
+          refreshKey,
+           setLoader,
+            ...props
+           }: any) {
+  const sizeMap = useRef<any>({})
```
[the remark this answers is not in the document]
Comment (dherault) [reply www/src/components/utils/SmoothScroller.tsx]: Linted, weird that the linter didn't catch it

File: www/src/components/clusters/Clusters.tsx (hunk 1 above)
Comment (dherault) [reply www/src/components/clusters/Clusters.tsx]: Rebased

**Review by dogmar — APPROVED** [review]
Success!

File: www/src/components/utils/SmoothScroller.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (dogmar) [reply www/src/components/utils/SmoothScroller.tsx]: Yes, very strange. ¯\_(ツ)_/¯

**Pull request merged**